composites/Plugin/Shared/components/YoastButton.js

```diff
@@ -39,18 +42,58 @@ export function addButtonStyles( component ) {
 			transform: translateY( 1px );
 			box-shadow: none;
 		}
+
+		// Only needed for Safari 10.
+		span {
+			display: inherit;
+			align-items: inherit;
+			justify-content: inherit;
+			width: 100%;
+		}
 	`;
 }
 
 /**
+ * Returns a component with a button and an inner span element.
+ *
+ * The inner span is only needed to fix a Safari 10 bug with flexbox and button
+ * elements. This bug is fixed in Safari Technology Previs and in the future it
+ * will be possible to remove this component and directly style a button element.
+ * See https://github.com/philipwalton/flexbugs#9-some-html-elements-cant-be-flex-containers
+ *
+ * @returns {ReactElement} The button with inner span.
+ */
+const YoastButtonBase = ( { className, onClick, type, children } ) => (
```

Comment:
Now you are limiting the amount of attributes that can be passed to the button. Isn't it better to deconstruct the props into the button? (`<button { ...props } >`)

Reply:
That will throw warnings like:
`Unknown props `backgroundColor`, `textColor`, `withTextShadow` on <button> tag.`